lib/Raven/Client.php

```diff
@@ -233,35 +238,49 @@ class Client
      */
     protected $_shutdown_function_has_been_set;
 
-    public function __construct($options_or_dsn = null, $options = [])
+    /**
+     * @var Configuration The client configuration
+     */
+    protected $config;
+
+    /**
+     * Constructor.
+     *
+     * @param Configuration $config The client configuration
+     */
+    public function __construct(Configuration $config)
```

Comment:
I actually pass the configuration class as an argument because it makes possible to extend it if any users wants to do it. If I create the instance inside the constructor then this won't be possible. Plus, my idea is to introduce a builder class so that users won't have the need to directly call `new Client(new Configuration())`. This is an example of what I imagine:

```php
$options = ['foo' => 'bar'];
$client = ClientBuilder::create($options)
    ->setTransportFactory(...)
    ->setLogger(...)
    ->setWhatheverCouldBeUseful(...)
    ->getClient();
```

I see that at this stage having to instantiate the Configuration class instead of directly passing the array of options feels bad, but we will anyway need to have a constructor that accepts more arguments as we expand the `Client` class to accept more arguments like the factories to create processors and transports. And this is not bad at all, because it means that users will have exact extensions points to extend the behaviour of Sentry without having to subclass the client for nearly everything they have to do now. Having a strict set of API is only a good thing, not a limitation. And finally, any framework will anyway abstract away how they create the client, so users will not need to interact with the arguments passed to the constructor. For libraries users, this is the worl and I never heard someone complaining because they had to pass an object instance instead of an array